search/django/weedid/tasks.py

```diff
@@ -115,3 +141,26 @@ def reindex_dataset(
         repository_dir=str(repository_dir),
         new_upload=False,
     )
+
+
+@shared_task
+def redeposit_dataset(
+    upload_id,
+    repository_dir=REPOSITORY_DIR,
+    download_dir=DOWNLOAD_DIR,
+    upload_dir=UPLOAD_DIR,
+):
+    upload_entity = Dataset.objects.get(upload_id=upload_id)
+    download_dir = Path(download_dir)
+    dataset_dir = Path(repository_dir) / upload_id
+    upload_user_dir = Path(upload_dir) / str(upload_entity.user_id)
```

Comment:
I am a little uncomfortable about reusing the original upload directory, but I suppose it is archival.